### PR TITLE
feat: honour .gitignore when linting filesystems

### LINKED ISSUE #166: honour `.gitignore` when linting filesystems
If we can support regex, we should always ignore those files present in the `.gitignore` file

In addition, we should add a new attribute to `.sasjslint` with the following example:

```
    "ignoreList":[
      "some/relative/directory",
      "/full/path",
      "relative/file.sas",
      "/path/to/file.sas"
    ]
```

`ignoreList` would contain a mix of files and directories.  For files, we only need to accept `.sas` as that is the only file type that we lint.  

The array items could be relative (to the `.sasjslint` file) or absolute.

### PR DESCRIPTION
**Pull request by sabhas**
## Issue

closes #166

## Intent

* add a new attribute `ignoreList` to LintConfig
* do not lint files that are specified in `.gitignore`

## Implementation

* added a new utility method `isIgnored` that specifies whether to ignore the linting of specified file/folder

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
- [x] All CI checks are green.
- [x] sasjslint-schema.json is updated with any new / changed functionality
- [x] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

File: src/utils/isIgnored.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { fileExists, readFile } from '@sasjs/utils'
+import path from 'path'
+import ignore from 'ignore'
+import { getLintConfig, getProjectRoot } from '.'
+import { LintConfig } from '../types'
+
+/**
+ * A function to check if file/folder path matches any pattern from .gitignore or ignoreList (.sasjsLint)
+ *
+ * @param fPath absolute path of file or folder
```
Comment (YuryShkoda) [inline src/utils/isIgnored.ts:10]: shouldn't it be `@param {string} fPath`?

File: src/utils/isIgnored.spec.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import * as fileModule from '@sasjs/utils/file'
+import * as getProjectRootModule from './getProjectRoot'
+import * as getLintConfigModule from './getLintConfig'
+import { DefaultLintConfiguration } from './getLintConfig'
+import { LintConfig } from '../types'
+import { isIgnored } from './isIgnored'
+
+describe('isIgnored', () => {
```
Comment (YuryShkoda) [inline src/utils/isIgnored.spec.ts:8]: we should also test if filepath e.g. `sasjs/common/init/init.sas` is ignored if there is an entry to ignore `sasjs/common`.

File: src/utils/isIgnored.ts (hunk 1 above)
Comment (sabhas) [reply src/utils/isIgnored.ts]: fixed

File: src/utils/isIgnored.spec.ts (hunk 2 above)
Comment (sabhas) [reply src/utils/isIgnored.spec.ts:7]: done

**Review by allanbowe — APPROVED** [review]

**Review by medjedovicm — APPROVED** [review]

**Pull request merged**